app/src/main/kotlin/io/eugenethedev/taigamobile/ui/screens/profile/ProfileScreen.kt

```diff
@@ -0,0 +1,337 @@
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.navigation.NavController
+import coil.annotation.ExperimentalCoilApi
+import coil.compose.rememberImagePainter
+import com.google.accompanist.insets.navigationBarsHeight
+import io.eugenethedev.taigamobile.R
+import io.eugenethedev.taigamobile.domain.entities.Project
+import io.eugenethedev.taigamobile.domain.entities.Stats
+import io.eugenethedev.taigamobile.domain.entities.User
+import io.eugenethedev.taigamobile.ui.components.appbars.AppBarWithBackButton
+import io.eugenethedev.taigamobile.ui.components.loaders.CircularLoader
+import io.eugenethedev.taigamobile.ui.theme.mainHorizontalScreenPadding
+import io.eugenethedev.taigamobile.ui.utils.ErrorResult
+import io.eugenethedev.taigamobile.ui.utils.LoadingResult
+import io.eugenethedev.taigamobile.ui.utils.subscribeOnError
+
+@Composable
+fun ProfileScreen(
+    navController: NavController,
+    onError: @Composable (message: Int) -> Unit = {},
+    userId: Long
+) {
+    val viewModel: ProfileViewModel = viewModel()
+    viewModel.onOpen(userId)
+
+    val currentUser by viewModel.currentUser.collectAsState()
+    currentUser.subscribeOnError(onError)
+    val currentUserStats by viewModel.currentUserStats.collectAsState()
+    currentUserStats.subscribeOnError(onError)
+    val currentUserProjects by viewModel.currentUserProjects.collectAsState()
+    currentUserProjects.subscribeOnError(onError)
+
+    ProfileScreenContentTest(
+        navigateBack = navController::popBackStack,
+        currentUser = currentUser.data,
+        currentUserStats = currentUserStats.data,
+        currentUserProjects = currentUserProjects.data ?: emptyList(),
+        isLoading = currentUser is LoadingResult || currentUserStats is LoadingResult || currentUserProjects is LoadingResult,
```

Comment:
I think this would look better:
```kotlin
listOf(currentUser, currentUserStats, currentUserProjects).any { it is LoadingResult }
```
Here and below